docs/proposals/dbaas/01-architecture/proposal.md

```diff
@@ -0,0 +1,695 @@
+# Kubernetes-native implementation for PMM/DBaaS
+
+## Summary
+
+PMM DBaaS has a lot of functionality to install/manage operators and CRUD operations for either a Percona XtraDB or a Percona Server For MongoDB cluster. However, the current implementation is not extensible enough and lacks scalability features.
+
+This proposal introduces architectural changes that are required to make DBaaS more extensible and more native for K8S.
+
+## Motivation
+
+The current architecture for DBaaS has the following components:
+
+1. PMM that exposes REST API for UI and some automation
+2. DBaaS controller that exposes gRPC API to PMM and works with Kubernetes clusters
+
+![Current DBaaS architecture](./current_arch.jpg)
+
+As a proof of concept, this architecture covers everything. However, it has the following issues:
+
+1. Non-native Kubernetes API to work with clusters. DBaaS controller has only gRPC API and this creates additional friction for community users to work/extend with the controller. One needs to generate a gRPC client to communicate with the DBaaS controller. Thus implementing the integration testing framework becomes a complex task to solve because test frameworks such as playwright or codecept.js do not have gRPC support.
+2. DBaaS controller has a set of CRUD endpoints for each database type (e.g. PXC and PSMDB). It adds additional room for bugs/inconsistencies and has the following issues:
+    * There’s no simple and generic API for any database cluster
+    * PMM needs to make two requests to get a list of created clusters (one for PSMDB and one for PXC clusters). In case we need to add support for another database, DBaaS controller would have an additional set of CRUD endpoints and PMM whould also call the list method for the new database type.
+
+3. DBaaS controller suffers from insufficient test coverage and integration testing because of the reasons mentioned above. Yet we can create an integration testing framework and increase coverage, but in that case it’ll require a substantial effort.
+4. Currently, DBaaS controller has only basic features such as CRUD operators for the database cluster and a lack of backup/restore features and additional configuration. There’s no way to specify additional parameters (Database configuration options, load balancer rules, storage class, backup schedule)
+5. REST API for PMM does not follow REST guidelines.
+
+Moving to OLM and a DBaaS operator will improve this situation.
+
+## Goals
```

Comment:
as I remember the current DBaaS implementation is limited to using the `default` namespace only in K8S. But there is nothing about using different namespaces in the list of goals